module/zfs/arc.c

```diff
@@ -7673,6 +7806,13 @@ arc_set_limits(uint64_t allmem)
 	/* How to set default max varies by platform. */
 	arc_c_max = arc_default_max(arc_c_min, allmem);
 }
+
+static inline size_t
+arc_ilog2(int a)
+{
+	return (a > 1 ? 1 + arc_ilog2(a >> 1) : 0);
```

Comment:
There is `highbit()` function, which should produce the similar result without recursion.